src/main.rs

```diff
@@ -115,8 +116,26 @@ async fn main() -> ExitCode {
         }
     };
 
+    if is_provisioning_complete(Some(&config), None) {
+        tracing::info!(
+            "Provisioning already completed earlier. Skipping provisioning."
+        );
+        return ExitCode::SUCCESS;
+    }
+
+    let clone_config = config.clone();
```

Comment:
FWIW I don't think this is "odd" or breaking any Rust conventions I'm aware of so doing this or doing it inline are both fine